jackson/src/main/java/io/gravitee/definition/jackson/datatype/services/core/deser/ScheduledServiceDeserializer.java

```diff
@@ -39,43 +41,61 @@ public ScheduledServiceDeserializer(Class<T> vc) {
     protected void deserialize(T service, JsonParser jsonParser, JsonNode node, DeserializationContext ctxt) throws IOException, JsonProcessingException {
         super.deserialize(service, jsonParser, node, ctxt);
 
-        // New version of scheduler service
-        final JsonNode triggerNode = node.get("trigger");
 
-        if (triggerNode != null) {
-            Trigger trigger = new Trigger();
-            final JsonNode rateNode = triggerNode.get("rate");
-
-            if (rateNode != null) {
-                trigger.setRate(rateNode.asLong());
-            } else {
-                throw ctxt.mappingException("[scheduled-service] Rate is required");
-            }
-
-            final JsonNode unitNode = triggerNode.get("unit");
-            if (unitNode != null) {
-                trigger.setUnit(TimeUnit.valueOf(unitNode.asText().toUpperCase()));
-            } else {
-                throw ctxt.mappingException("[scheduled-service] Unit is required");
-            }
-            service.setTrigger(trigger);
+        final JsonNode scheduleNode = node.get("schedule");
+        String schedule = null;
+        if (scheduleNode != null) {
+            schedule = scheduleNode.asText();
         } else {
-            // Ensure backward compatibility
-            final JsonNode intervalNode = node.get("interval");
 
-            if (intervalNode != null) {
-                Trigger trigger = new Trigger();
-                trigger.setRate(intervalNode.asLong());
+            Long rate = null;
+            TimeUnit unit = null;
+            if (node.has("trigger")) {
+                final JsonNode triggerNode = node.get("trigger");
+                final JsonNode rateNode = triggerNode.get("rate");
+
+                if (rateNode != null) {
+                    rate = rateNode.asLong();
+                } else {
+                    throw ctxt.mappingException("[scheduled-service] Rate is required");
```

Comment:
it is deprecated, do you know if we have a better choice ?

Reply:
Perhaps https://javadoc.io/doc/com.fasterxml.jackson.core/jackson-databind/2.9.8/com/fasterxml/jackson/databind/DeserializationContext.html#instantiationException-java.lang.Class-java.lang.String-

Reply:
> Note that most of the time this method should NOT be called; instead, handleMissingInstantiator(java.lang.Class<?>, com.fasterxml.jackson.databind.deser.ValueInstantiator, com.fasterxml.jackson.core.JsonParser, java.lang.String, java.lang.Object...) should be called